arrow/locales.py

```diff
@@ -3219,20 +3245,19 @@ def _format_timeframe(self, timeframe: TimeFrameLiteral, delta: int) -> str:
 
         if delta == 0:
             key = "zero"  # And *never* use 0 in the singular!
-        elif delta > 0:
-            key = "future"
-        else:
+        elif delta < 0:
             key = "past"
-        form: Union[List[str], str] = form[key]
-
-        if isinstance(form, list):
-            if 2 <= abs_delta % 10 <= 4 and (
+        else:
+            if "future-singular" not in form:
+                key = "future"
+            elif 2 <= abs_delta % 10 <= 4 and (
```

Comment:
Can you just explain this part?

Reply:
Some timeframe units contain two future tenses, future-singular and future-plural. Timeframe units without these two forms just have the form "future". This check is with regards to that.